packages/catalog-process/src/services/catalogService.ts

```diff
@@ -218,7 +218,14 @@ export function catalogServiceBuilder(
       const eService = await readModelService.getEServiceById(eServiceId);
 
       await repository.createEvent(
-        updateEserviceLogic({ eService, eServiceId, authData, eServiceSeed })
+        await updateEserviceLogic({
+          eService,
+          eServiceId,
+          authData,
+          eServiceSeed,
+          getEServiceByNameAndProducerId:
+            readModelService.getEServiceByNameAndProducerId,
```

Comment:
```suggestion
          eServiceDuplicate:
            await readModelService.getEServiceByNameAndProducerId({
              name: eServiceSeed.name,
              producerId: authData.organizationId,
            }),
```
Why not doing something like this? 
So you don't have to bring that function further and you can check if `eServiceDuplicate` is `undefined`

Reply:
Updated

Reply:
cc @galales

Reply:
Update: I reverted the change to follow the suggestion raised in the thread below. Now the `getEServiceByNameAndProducerId` function is passed to the `_Logic` function.